host/src/server/api/v2/proof/mod.rs

```diff
@@ -64,42 +64,46 @@ async fn proof_handler(
 
     let mut manager = prover_state.task_manager();
     let status = manager.get_task_proving_status(&key).await?;
-
-    let Some((latest_status, ..)) = status.0.last() else {
-        // If there are no tasks with provided config, create a new one.
-        manager.enqueue_task(&key).await?;
-
-        prover_state
-            .task_channel
-            .try_send(Message::from(&proof_request))?;
-
-        return Ok(TaskStatus::Registered.into());
-    };
-
-    match latest_status {
-        // If task has been cancelled add it to the queue again
-        TaskStatus::Cancelled
-        | TaskStatus::Cancelled_Aborted
-        | TaskStatus::Cancelled_NeverStarted
-        | TaskStatus::CancellationInProgress => {
-            manager
-                .update_task_progress(key, TaskStatus::Registered, None)
-                .await?;
+    match status.0.last() {
+        Some((latest_status, ..)) => {
+            match latest_status {
+                // If task has been cancelled
+                TaskStatus::Cancelled
+                | TaskStatus::Cancelled_Aborted
+                | TaskStatus::Cancelled_NeverStarted
+                | TaskStatus::CancellationInProgress
+                // or if the task is failed, add it to the queue again
+                | TaskStatus::GuestProverFailure(_)
+                | TaskStatus::UnspecifiedFailureReason => {
+                    manager
+                        .update_task_progress(key, TaskStatus::Registered, None)
+                        .await?;
```

Comment:
If the update fails or `task_channel` is full, do we need to retry?

Reply:
both update fail & task_channel error will not be log into taskdb. For channel error, when client retries, if that time channel is free, it should be ok.
but update failure may not automatically recovered I think. for example if redis has no response, we must restart k8s node??